src/Kralizek.Extensions.Configuration.AWSSecretsManager/Internal/SecretsManagerConfigurationProvider.cs

```diff
@@ -215,7 +214,9 @@ private async Task<HashSet<(string, string)>> FetchConfigurationAsync(Cancellati
                 {
                     if (!Options.SecretFilter(secret)) continue;
 
-                    var secretValue = await Client.GetSecretValueAsync(new GetSecretValueRequest { SecretId = secret.ARN }, cancellationToken).ConfigureAwait(false);
+                    var request = new GetSecretValueRequest { SecretId = secret.ARN };
+                    Options.ConfigureSecretValueRequest(request, new GetSecretValueRequestContext(secret));
```

Comment:
```suggestion
                    Options?.ConfigureSecretValueRequest(request, new GetSecretValueRequestContext(secret));
```

Just to be safe.

Reply:
I went with `Options.ConfigureSecretValueRequest?.Invoke(request, new SecretValueContext(secret));` because it's the action that might be null, not the options.

Reply:
Good spot. That's what i meant :)